### ISSUE TITLE
Vague error message when matchmaking function service unreachable

### ISSUE THREAD

**Issue by bfulljes** [googleforgames/open-match#1153]
**What happened**:

I followed the https://open-match.dev/site/docs/tutorials/matchmaker101/ tutorial.  I briefly switched around the names of some things to match a naming convention for our project as a whole, not thinking it would cause any issue.  Unfortunately I slipped and misnamed the matchmaking function service endpoint in the director config.

When everything started, I began to see the following errors spammed in batches on the `om-backend` process:

```
time="2020-03-11T04:44:08Z" level=error msg="failed to run match function for profile" app=openmatch component=app.backend error="rpc error: code = Canceled desc = context canceled"
time="2020-03-11T04:44:08Z" level=error msg="error(s) in FetchMatches call." app=openmatch component=app.backend mmfErr="rpc error: code = Canceled desc = context canceled" syncErr="<nil>"
```

**What you expected to happen**:

I would have been able to fix my incorrect configuration much more quickly if the error had said anything about a connection issue, such as connection refused or unable to connect.

**How to reproduce it (as minimally and precisely as possible)**:

Follow the matchmaker 101 tutorial, but change [this line](https://github.com/googleforgames/open-match/blob/release-0.9/tutorials/matchmaker101/solution/director/main.go#L37) to something like `mm101-nope.mm101-tutorial.cluster.svc.local` for the endpoint value.

**Anything else we need to know?**:

**Output of `kubectl version`**:

```
Client Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.5", GitCommit:"20c265fef0741dd71a66480e35bd69f18351daea", GitTreeState:"clean", BuildDate:"2019-10-15T19:16:51Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.5", GitCommit:"20c265fef0741dd71a66480e35bd69f18351daea", GitTreeState:"clean", BuildDate:"2019-10-15T19:07:57Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"linux/amd64"}
```

**Cloud Provider/Platform (AKS, GKE, Minikube etc.)**:  Docker Desktop for Mac

**Open Match Release Version**: 0.9.0

**Install Method(yaml/helm):**: Helm

**Comment by Laremere** [comment]
I suspect what is happening here is that the function run call is trying to succeed with no timeout, and the synchronizer is canceling the call.  How to get a good error out of it, I'm unsure.  Maybe we should be wrapping the error indicating the call was never started.

Also, we're logging and also returning, which is making this error show up, when it's not the root error returned to the caller.  This adds to the confusion.

**Comment by akremsa** [comment]
@Laremere I think this issue is related to #1102 and there is a PR #1255

**Comment by castaneai** [comment]
The Unary Interceptor implemented in #1255 does not solve this problem, because FetchMatches RPC is a Server streaming RPC.
Don't we need a Streaming Interceptor?

**Comment by syntxerror** [comment]
Closing due to root issue being closed. A streaming interceptor for matches return  is separate from the current issue that was addressed. If we believe that a streaming interceptor is needed, we should open a new issue to implement it. Sound good @castaneai ?
